libraries/AP_HAL_ChibiOS/hwdef/FlywooF405Pro/hwdef.dat

```diff
@@ -0,0 +1,151 @@
+
+# hw definition file for processing by chibios_hwdef.py
+# for FLYWOOF405PRO hardware.
+# thanks to betaflight for pin information
+
+# MCU class and specific type
+MCU STM32F4xx STM32F405xx
+
+# board ID for firmware load
+APJ_BOARD_ID 1137
+
+# crystal frequency, setup to use external oscillator
+OSCILLATOR_HZ 8000000
+
+FLASH_SIZE_KB 1024
+
+# bootloader takes first sector
+FLASH_RESERVE_START_KB 48
+
+define HAL_STORAGE_SIZE 16384
+define STORAGE_FLASH_PAGE 1
+
+STM32_ST_USE_TIMER 4
+define CH_CFG_ST_RESOLUTION 16
+
+# SPI devices
+
+# SPI1
+PA5 SPI1_SCK SPI1
+PA6 SPI1_MISO SPI1
+PA7 SPI1_MOSI SPI1
+
+# SPI3
+PC10 SPI3_SCK SPI3
+PC11 SPI3_MISO SPI3
+PC12 SPI3_MOSI SPI3
+
+# Chip select pins
+PB3 FLASH1_CS CS
+PB14 OSD1_CS CS
+PB12 GYRO1_CS CS
+
+# Beeper
+PC13 BUZZER OUTPUT GPIO(80) LOW
+define HAL_BUZZER_PIN 80
+define HAL_BUZZER_ON 1
+define HAL_BUZZER_OFF 0
+
+# SERIAL ports
+SERIAL_ORDER OTG1 USART1 USART2 USART3 UART4 UART5 USART6
+# PA10 IO-debug-console
+PA11 OTG_FS_DM OTG1
+PA12 OTG_FS_DP OTG1
+
+# USART1
+PA10 USART1_RX USART1 NODMA
+PB6 USART1_TX USART1 NODMA
+
+# USART2 (ESC Telemetry)
+PD6 USART2_RX USART2 NODMA
+DEFAULT_SERIAL2_PROTOCOL SerialProtocol_ESCTelemetry
+
+# USART3 (DJI)
+PB10 USART3_TX USART3
+PB11 USART3_RX USART3
+DEFAULT_SERIAL3_PROTOCOL SerialProtocol_DJI_FPV
+
+# UART4 (SBUS)
+PA0 UART4_TX UART4
+PA1 UART4_RX UART4 NODMA ALT(1)
```

Comment:
If we cant get DMA on RX4 then isn't it worthless for CRSF and ELRS? that means there are no options for those since UART1 cant be used, UART3 is only on the DJI connector and cant be used if DJI telem is in use, leaving only UART6 which is targetted for GPS....so you cant have GPS+CRSF/ELRS+DJI...I would think sacrificing UART3 DMA vs allowing RX5 to work on SBUS and having RX4 be DMA capable would worth attempting....

Reply:
I actually spent a long time trying to exactly this and failed. Nothing you give up makes for a substitute to give another UART DMA. You don't strictly need DMA for GPS - you can use the option bit to run at 115k so it would be possible to put CRSF on UART6 and GPS on UART1 or UART4.

Reply:
going to take a lot of words in the wiki to explain how someone can use this, especially if they are migrating from an existing use...

Reply:
Ok I have changed the defaults so that GPS should just work on UART1